src/Data/Traversable/Sort/Vector.hs

```diff
@@ -0,0 +1,49 @@
+{-# LANGUAGE Rank2Types          #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+module Data.Traversable.Sort.Vector (sortTraversableBy, sortTraversable) where
+import           Control.Monad.ST.Strict
+import           Control.Monad.State.Strict
+import           Data.Foldable
+import qualified Data.Vector.Algorithms.Intro as Intro
+import qualified Data.Vector.Mutable          as VM
+
+{-# INLINE sortTraversableBy #-}
```

Comment:
This name is confusing. This is not actually a sorting function at all, and the `Ord` constraint isn't needed.